[sethteo] iP

sethteo opened the pull request:
# Slaybot
> "**Slaayyyyy**"

Slaybot frees your mind of having to remember things you need to do. It's,

- Text-based
- Easy to learn /j 😉
- Super fast to use /s 😉

Very very key features:

- Managing tasks
- Saving tasks
- Task searching

All you need to do is,

1. Clone it from [here](https://github.com/sethteo/ip/tree/master)
2. Run it

To be added features:

- [x] GUI
- [ ] Task reminders



If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```
public class Launcher {
    public static void main(String[] args) {
        Application.launch(Main.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

sethteo committed:
Level-0: Rename, Greet, Exit

sethteo committed:
Add Level 0 Tag

sethteo committed:
Level-1

sethteo committed:
Level-2

sethteo committed:
Level-3

sethteo committed:
Level-4

sethteo committed:
Added automated ui testing

sethteo committed:
Added exception handling

sethteo committed:
Added delete function

sethteo committed:
Added file saving feature

sethteo committed:
Added file saving feature

sethteo committed:
Add level 7 branch

sethteo committed:
Datetime functionality for Deadline

sethteo committed:
Datetime functionality for Event

sethteo committed:
Added more OOP

sethteo committed:
Added OOP Classes

sethteo committed:
Added more OOP

sethteo committed:
Added packages

sethteo committed:
Merge remote-tracking branch 'origin/add-gradle-support'

sethteo committed:
Added JUnit

sethteo committed:
Added JUnit

sethteo committed:
Add Javadoc for Storage

sethteo committed:
Add Javadoc for TaskList

sethteo committed:
Add Javadoc for Ui

sethteo committed:
Add Javadoc for Parser

sethteo committed:
Add Javadoc for Deadline

sethteo committed:
Add Javadoc for Event

sethteo committed:
Add Javadoc for Task

sethteo committed:
Add Javadoc for ToDo

sethteo committed:
Update coding style

sethteo committed:
Add find feature

sethteo committed:
Merge A-CodingStandard

sethteo committed:
Merge Level-9

sethteo committed:
Fix checkstyle

sethteo commented:
Slaybot
"Slaayyyyy"

Slaybot frees your mind of having to remember things you need to do. It's,

- Text-based
- Easy to learn /j
- Super fast to use /s


Very very key features:

- Managing tasks
- Saving tasks
- Task searching

To be added features:

- GUI
- Task reminders

sethteo committed:
Add Checkstyle

sethteo committed:
Add GUI

sethteo committed:
Add styling to dialog boxes

sethteo committed:
Implement logic into GUI

sethteo committed:
Fix dialog box

c-wenlong commented on src/main/java/entity/Deadline.java:
```diff
@@ -0,0 +1,44 @@
+            return "D | Not Done | " + this.title;
+        }
+    }
+    @Override
```
Maybe you could add an empty line between methods for consistency. :)

c-wenlong commented on src/main/java/entity/Task.java:
```diff
@@ -0,0 +1,33 @@
+ * Abstract class which would be inherited by the various subclasses Deadline, ToDo and Event
+ */
+public abstract class Task {
+    protected String title;
```
Why is the title and marked here protected? whereas deadline and event class fields are private? Perhaps could standardise? Also perhaps it might be more clear to name boolean with is___, in this case, isMarked? :)

c-wenlong commented on src/main/java/entity/Task.java:
```diff
@@ -0,0 +1,33 @@
+        return this.marked;
+    }
+
+    public void setMarked(boolean bool) {
```
It might be better to give an intentional name rather than bool here.

sethteo replied:
Good suggestion :)

c-wenlong commented on src/main/java/entity/Task.java:
```diff
@@ -0,0 +1,33 @@
+        this.marked = bool;
+    }
+
+    public abstract String save();
```
Good use of abstract method to ensure method is implemented in the child classes! 👍

c-wenlong commented on src/main/java/slaybot/DialogBox.java:
```diff
@@ -0,0 +1,75 @@
+ */
+public class DialogBox extends HBox {
+    @FXML
+    private Label dialog;
```
Perhaps more description could be provided for the fields and methods in this class? Even better with some inline documentation with methods that has some level of abstraction. :)

c-wenlong commented on src/main/java/slaybot/Main.java:
```diff
@@ -0,0 +1,30 @@
+ */
+public class Main extends Application {
+
+    private Slaybot slaybot = new Slaybot();
```
Same here, could use some JavaDoc. :)

c-wenlong commented on src/main/java/slaybot/MainWindow.java:
```diff
@@ -0,0 +1,50 @@
+ */
+public class MainWindow extends AnchorPane {
+    @FXML
+    private ScrollPane scrollPane;
```
Could use some JavaDoc here as well :)

c-wenlong commented on src/main/java/slaybot/Parser.java:
```diff
@@ -0,0 +1,176 @@
+ */
+public class Parser {
+
+    enum Command {
```
Good use of enums here to represent commands, something that I can learn from you :)

c-wenlong commented on src/main/java/slaybot/Slaybot.java:
```diff
@@ -0,0 +1,52 @@
+ */
+public class Slaybot {
+
+    private TaskList tasks;
```
Could use some JavaDoc here :)

c-wenlong commented on src/main/java/slaybot/TaskList.java:
```diff
@@ -0,0 +1,99 @@
+        this.tasks.add(t);
+    }
+
+    /**
```
Unused code could be deleted for a more elegant codebase :)

sethteo replied:
Will clean up my code as well as add more JavaDocs, thank you! @c-wenlong

c-wenlong reviewed (commented):
Overall, the code is very readable and good use of OOP structures such as polymorphism and encapsulation. Your classes are very well defined too. I see that you have also personalised your chatbot, kudos to being creative! The only thing is to document your code a bit more to enhance understanding. Great work!

sethteo committed:
Add Assertions

Added Java assertions to code
-start method in Ui class
The command variable might be null

Added assert to ensure that the command enumeration is not null before the switch statement.

This is done to prevent any NullPointerExceptions

-markTask method in TaskList class
The changes to the target Task might not be changed.

Added assert to ensure that the marked status of the target Task is equal to the intended boolean.

This is done to prevent any logical errors that might occur.

-unmarkTask method in TaskList class
The changes to the target Task might not be changed.

Added assert to ensure that the marked status of the target Task is equal to the intended boolean.

This is done to prevent any logical errors that might occur.

sethteo committed:
Fix code quality

Code contained many unused methods as well as
 commented out code blocks.

This cause the code to be messy.

- Removed the unused variables and methods
- Removed the commented out code blocks
- Changed the boolean variables to sound more like
booleans e.g marked to isMarked
- Changed the naming of variables to make more sense
e.g bool to markedStatus

This is done to follow the code quality standards

sethteo committed:
Merge pull request #1 from sethteo/branch-A-Assertions

Add Assertions

sethteo committed:
Add Sort Feature

Tasks are currently displayed on FIFO basis.

This results in a less than optimal user experience due to the
Tasks not being sorted by their deadline dates.

Sorting is done by placing tasks with earlier end dates at the top
of the list while tasks with later/no end dates (e.g ToDo Tasks) at
the bottom. This sorting is done by using a custom TaskComparator
class which does the sorting following the above requirements.

A TaskComparator is preffered so as to capitalise on the List.sort()
method.

sethteo committed:
Merge branch 'master' into branch-A-CodeQuality

sethteo committed:
Merge pull request #2 from sethteo/branch-A-CodeQuality

Fix code quality

sethteo committed:
Merge branch 'master' into branch-C-Sort

sethteo committed:
Merge pull request #3 from sethteo/branch-C-Sort

Add Sort Feature

sethteo committed:
Update build.gradle file

sethteo committed:
Fix checkstyle

sethteo committed:
Fix file storage for Jar

sethteo committed:
Fix SLAP for Ui class

sethteo committed:
Fix Deadline date bug

sethteo committed:
Update README.md

sethteo committed:
Update README.md

sethteo committed:
Update README.md

sethteo committed:
Update README.md

sethteo committed:
Update README.md

sethteo committed:
Fix checkstyle

sethteo committed:
Update README.md

sethteo committed:
Fix README.md

sethteo committed:
Fix data loading bug

sethteo committed:
Fix checkstyle

sethteo committed:
Update JUnit

sethteo committed:
Update README.md
